### PR TITLE
Support collection grouping

### PR DESCRIPTION
**Pull request by hcorta**
PR for collections-grouping support 🙂
I'd willing to hear suggestions on code style or similar.

### CODE REVIEW

**Comment by normdoow** [comment]
Thanks @hcorta will be looking at this over the next couple of days.

File: example/next.config.js
Comment (normdoow) [inline example/next.config.js:13]: Can we reset this back to what it was?

File: example/pages/[pageId].tsx
Comment (normdoow) [inline example/pages/[pageId].tsx:32]: can we reset this back as well?

**Comment by normdoow** [comment]
Could you make your notion page at `6fa923939ba54d7583612a68a6467e34` able to be duplicated so I can add it to our Notion test suite at https://travis-fischer.notion.site/Notion-Kit-Test-Suite-067dd719a912471ea9a3ac10710e7fdf? Thanks!

**Comment by normdoow** [comment]
Doing some testing. I'm not seeing my two rows in group 1 and group 2.
image
Testing this page: https://travis-fischer.notion.site/dd5d904fb13d402cbba6d6bdfecfa181?v=83a14203fe024932b711767c22819b39
Trying to figure out why.

**Review by normdoow — CHANGES_REQUESTED** [review]
Just a couple of things. Code looks good! Just need to make sure its working on the case I found. Thanks!

**Comment by hcorta** [comment]
@normdoow sure, let me review those things. I noticed as well that when selecting the "Hide empty groups" option, the hidden property is not explicitly received. I'll check on that.

**Comment by hcorta** [comment]
@normdoow a few things:
* I restored default values as required.
* I created this template, hope that helps: https://hcorta.notion.site/Collection-Grouping-671539f6022c4327a1f6de8069db3abe
* I could not solve the "Hide empty groups" issue. I could not find anywhere (collectionData, collectionView, etc) the info from where I could filter those groups.
* I could not replicate your case, may you give me more info? 
* The board collections needs to be reviewed when a second groupBy is applied, but that could be motive for another PR?

**Comment by normdoow** [comment]
> @normdoow a few things:
> 
> * I restored default values as required.
> * I created this template, hope that helps: https://hcorta.notion.site/Collection-Grouping-671539f6022c4327a1f6de8069db3abe
> * I could not solve the "Hide empty groups" issue. I could not find anywhere (collectionData, collectionView, etc) the info from where I could filter those groups.
> * I could not replicate your case, may you give me more info?
> * The board collections needs to be reviewed when a second groupBy is applied, but that could be motive for another PR?

All good. I think we will be ready to merge this in once we make sure the issue I was seeing where the rows aren't showing isn't an issue. We can wait for another PR for the board collections.

So testing on your notion template, I get the no description list rows but not the rows for the other groups.
image
Looking into the returned data. For some reason that data isn't coming back from the notion-client query endpoint.
image
The `blockIds` is an empty array.
Pretty sure I have your fork pulled correctly to test from it. Trying to think what else this difference could be from.

**Comment by hcorta** [comment]
@normdoow I see. It seems like I missed something for multi-select groupings.... I'll check on that. Thanks for investigating a little further the problem 🙂

**Comment by hcorta** [comment]
@normdoow I thought about adding a prop to the **NotionRenderer**, _**hideEmptyGroups**_ for filtering groups with no children. Would it make sense?

**Comment by normdoow** [comment]
> @normdoow I thought about adding a prop to the **NotionRenderer**, _**hideEmptyGroups**_ for filtering groups with no children. Would it make sense?

Hmm not sure if we need that. Notion allows that as an option correct, but by default it shows empty groups. So I'd say lets just let it be the default for now.

**Comment by hcorta** [comment]
@normdoow I solved the issue with the multi_select grouping. It should be all ok now.

**Comment by normdoow** [comment]
Hey @hcorta, I haven't been able to complete reviewing this. Been off on vacation. I'll be able to get to it hopefully in a couple of days.

**Comment by normdoow** [comment]
@hcorta, everything is looking good. The only thing that I am seeing is that not all the data for boards view is showing up.
I'm fine with not having board views work with multiple groupBys for this PR. But we should be able to get it to work with one as it was perviously.
This is the page id I was seeing the issue on: ba528cc0a8dd41bcb74c7e9f80a4785d
On Notion: 
image
On react-notion-x:
image

**Comment by normdoow** [comment]
@hcorta looks like we are getting the correct data for the board. Just not showing it correctly. I'm looking into it right now to see if I can fix it.

**Comment by normdoow** [comment]
I figured out the issue.
image
In collection-view-board.tsx

**Review by normdoow — APPROVED** [review]
Great job on this @hcorta. Thanks for the help. I'll that fix after merge.

**Comment by normdoow** [comment]
Live on v4.13.0 💪

**Comment by hcorta** [comment]
Thanks @normdoow! Glad to help 😊

**Comment by adlpz** [comment]
Hi! I haven't looked to much into it but I've found the Gallery view fails in 4.14.2 with `TypeError: Cannot read property 'blockIds' of undefined`. The stack trace refers to `node_modules/react-notion-x/build/cjs/components/collection-view-gallery.js:31:166` and I have feeling it's related to this change.

Reverting to 4.11 fixed the issue.

I'm writing just to give the heads up in case it's something trivial that was missed. Otherwise, if I have some time, I'll look into it.

**Comment by hcorta** [comment]
Hi @adlpz, could you share the Notion page you're trying to render?

**Comment by adlpz** [comment]
@hcorta Hey, it was a simple page with an inline Gallery database, just for testing. Here's the public link: https://prealfa.notion.site/Blog-d99da35d2fbd42f189a9ad74c158aee7

**Pull request merged**
